sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs

```diff
@@ -33,15 +33,16 @@ public override bool Equals(object obj)
         /// <param name="left">The first identifier to compare.</param>
         /// <param name="right">The second identifier to compare.</param>
         /// <returns>True if the types and <see cref="RawId"/> match.</returns>
-        public static bool operator ==(CommunicationIdentifier left, CommunicationIdentifier right) => left.GetType() == right.GetType() && Equals(left, right);
+        public static bool operator ==(CommunicationIdentifier left, CommunicationIdentifier right)
+            => ReferenceEquals(left, right) || left is not null && right is not null && Equals(left, right);
```

Comment:
Not with this fix, but I notice a few classes accessing left.Equals() where left can be null itself. I think this case can cause an exception too.

Reply:
Do you have an example where `left.Equals` is being called?

Reply:
@mikehang-msft Resolving to complete this PR, let me know if you encounter any unsafe occurrence and we can follow up.

Reply:
Sorry for a late response. Yes, we should resolve this PR and tackle other issues later. 

I did a quick search on Azure.Communication.MediaComposition project (there could be more in the solution). The following classes have left.Equals() calls. They are all auto-gen and implementing IEquatable<> interface.

        public static bool operator ==(AudiencePosition left, AudiencePosition right) => left.Equals(right);	 mikehang-msft\azure-sdk-for-net\sdk\communication\Azure.Communication.MediaComposition\src\Generated\Models\AudiencePosition.cs	39	28
        public static bool operator ==(CompositionStreamState left, CompositionStreamState right) => left.Equals(right);	 mikehang-msft\azure-sdk-for-net\sdk\communication\Azure.Communication.MediaComposition\src\Generated\Models\CompositionStreamState.cs	36	28
        public static bool operator ==(LayerVisibility left, LayerVisibility right) => left.Equals(right);	 mikehang-msft\azure-sdk-for-net\sdk\communication\Azure.Communication.MediaComposition\src\Generated\Models\LayerVisibility.cs	33	28
        public static bool operator ==(LayoutType left, LayoutType right) => left.Equals(right);	 mikehang-msft\azure-sdk-for-net\sdk\communication\Azure.Communication.MediaComposition\src\Generated\Models\LayoutType.cs	42	28
        public static bool operator ==(MediaInputType left, MediaInputType right) => left.Equals(right);	 mikehang-msft\azure-sdk-for-net\sdk\communication\Azure.Communication.MediaComposition\src\Generated\Models\MediaInputType.cs	57	28
        public static bool operator ==(MediaOutputType left, MediaOutputType right) => left.Equals(right);	 mikehang-msft\azure-sdk-for-net\sdk\communication\Azure.Communication.MediaComposition\src\Generated\Models\MediaOutputType.cs	42	28
        public static bool operator ==(RtmpMode left, RtmpMode right) => left.Equals(right);	 mikehang-msft\azure-sdk-for-net\sdk\communication\Azure.Communication.MediaComposition\src\Generated\Models\RtmpMode.cs	33	28
        public static bool operator ==(SupportPosition left, SupportPosition right) => left.Equals(right);	 mikehang-msft\azure-sdk-for-net\sdk\communication\Azure.Communication.MediaComposition\src\Generated\Models\SupportPosition.cs	39	28